go/mysql/flavor_filepos.go

```diff
@@ -207,6 +207,16 @@ func (flv *filePosFlavor) status(c *Conn) (SlaveStatus, error) {
 		file: resultMap["Relay_Master_Log_File"],
 		pos:  pos,
 	}
+
+	relayPos, err := strconv.Atoi(resultMap["Relay_Log_Pos"])
+	if err != nil {
+		return SlaveStatus{}, fmt.Errorf("invalid FilePos GTID (%v): expecting pos to be an integer", resultMap["Exec_Master_Log_Pos"])
+	}
+
+	status.RelayLogPosition.GTIDSet = filePosGTID{
+		file: resultMap["Relay_Log_File"],
+		pos:  relayPos,
+	}
```

Comment:
I don't know if it is useful to duplicate Position and RelayLogPosition into FilePosition and FileRelayLogPosition for this flavor. Something to keep in mind as you get into using these structs to build logic.

Reply:
I was wondering the same thing actually. I decided not too because the information is already there for us. I'm still on the fence about the right approach here.

Reply:
The goal is to make it possible to write ERS without knowing the underlying flavor. If we can do that, it's a sign we've done a good job of abstracting the per-flavor logic.

For the filepos flavor, this information is already in another field, but you don't know that from looking at a `SlaveStatusResponse` unless you know the flavor. To avoid leaking this flavor-specific logic outside the flavor interface, we should always fill in these new fields for every flavor.

Reply:
That's a very good point that I hadn't considered. Ok, will fill in right now.

Reply:
Done.